cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -1240,10 +1282,10 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
                     }
                 ];
                 if (xblockInfo.isChapter()) {
-                    tabs[0].editors = [ReleaseDateEditor];
+                    tabs[0].editors = [ReleaseDateEditor, OptionalCompletionEditor];
                     tabs[1].editors = [StaffLockEditor];
                 } else if (xblockInfo.isSequential()) {
-                    tabs[0].editors = [ReleaseDateEditor, GradingEditor, DueDateEditor];
+                    tabs[0].editors = [ReleaseDateEditor, GradingEditor, DueDateEditor, OptionalCompletionEditor];
```

Comment:
Regarding [this comment](https://github.com/openedx/edx-platform/pull/34275#discussion_r1506686792) about hiding the editor when the completion tracking is not enabled, please see the following parts of code that handle a similar scenario for another editor:

https://github.com/open-craft/edx-platform/blob/a8535ffaeaf44000f46597c7663c950065d0d3eb/cms/templates/base.html#L160

https://github.com/open-craft/edx-platform/blob/42451caa2612a63b42bf5686e6217817e0dd2098/cms/static/js/views/modals/course_outline_modals.js#L1290-L1292

I think we shouldn't move this to a follow-up ticket because this feature is not enabled by default in `edx-platform`.

Reply:
wow i didn't realize it'd be so straight forward. implementing this

Reply:
done. I wasn't able to test it manually without hardcoding the value in the base.html. it seems waffle flags are cached somewhere and take quite some time to invalidate the cache. 

In case you want to test it and have my same problem you can hard code  `completion_tracking_enabled` to `False` in `cms/templates/base.html`  line 162.